### PR TITLE
Simplify the bconv2d padding functor.

### PR DESCRIPTION
**Pull request by AdamHillier**
## What do these changes do?

This PR simplifies the `PaddingFunctor` implementation by making all three methods static and removing some arguments that were not being used in the method bodies.

## How Has This Been Tested?

CI.

## Benchmark Results

CI.

## Related issue number

~~This is a draft as it depends on #514.~~

### CODE REVIEW

File: larq_compute_engine/core/bconv2d/zero_padding_correction.h [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+ * at the edge of the output tensor.
+ *
+ * The correction for a given output element depends on the values of the
+ * filters and the position in the output tensor. However, as the filter values
+ * are static we can cache some of the computation; this is done in
+ * `CacheCorrectionValues`.
+ */
+
+#ifndef COMPUTE_ENGINE_CORE_BCONV2D_PADDING_FUNCTOR_H_
+#define COMPUTE_ENGINE_CORE_BCONV2D_PADDING_FUNCTOR_H_
```
Comment (lgeiger) [inline larq_compute_engine/core/bconv2d/zero_padding_correction.h:19]:
```suggestion
#ifndef COMPUTE_ENGINE_CORE_BCONV2D_ZERO_PADDING_CORRECTION_H_
#define COMPUTE_ENGINE_CORE_BCONV2D_ZERO_PADDING_CORRECTION_H_
```

File: larq_compute_engine/core/bconv2d/zero_padding_correction.h [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+  }      // batch
+  return;
+}
+
+}  // namespace zero_padding_correction
+}  // namespace bconv2d
+}  // namespace core
+}  // namespace compute_engine
+
+#endif  // COMPUTE_ENGINE_CORE_BCONV2D_PADDING_FUNCTOR_H_
```
Comment (lgeiger) [inline larq_compute_engine/core/bconv2d/zero_padding_correction.h:304]:
```suggestion
#endif  // COMPUTE_ENGINE_CORE_BCONV2D_ZERO_PADDING_CORRECTION_H_
```

**Review by lgeiger — CHANGES_REQUESTED** [review]
Nice cleanup.

**Review by lgeiger — APPROVED** [review]

**Review by Tombana — APPROVED** [review]
Very clean :broom:

**Pull request merged**